site/content/languages/go.md

```diff
@@ -241,6 +241,73 @@ signature unrelated to the source, more care is needed implementing the host
 side, to ensure the proper count of parameters are read and results written to
 the Go stack.
 
+## Hacking
+
+If you run into an issue where you need to change Go's sourcecode, the first
+thing you should do is read the [contributing guide][20], which details how to
+confirm an issue exists and a fix would be accepted. Assuming they say yes, the
+next step is to ensure you can build and test go.
+
+### Make a branch for your changes
+
+First, clone upstream or your fork of golang/go and make a branch off `master`
+for your work, as GitHub pull requests are against that branch.
+
+```bash
+$ git clone https://github.com/golang/go.git
+$ cd go
+$ git checkout -b my-fix
+```
+
+### Build a branch-specific `go` binary
+
+While your change may not affect the go binary itself, there are checks inside
+go that require version matching. Build a go binary from source to avoid these:
+
+```bash
+$ cd src
+$ GOOS=js GOARCH=wasm ./make.bash
```

Comment:
Perhaps it differs too much from the norm to suggest, but JFYI, I didn't have xcode tools set up yet it seems, but following the error message's suggestion of `CGO_ENABLED=0` worked, and the below `syscall/js` test ran fine. But I don't know what problems it might cause in other development (race detector uses cgo so won't work with this build, not sure what else).

Reply:
just in case it helps others, this and other things messed up for me today when I updated OS/x until I did `xcodebuild -runFirstLaunch`